### ISSUE TITLE
Error open RocksDB database in docker

### ISSUE THREAD

**Issue by kaonick** [apache/pulsar#3031]
Below is what I run in windows10's dos command

`D:\temp>docker run -it -p 6650:6650 -p 8080:8080 -v d:/temp/data:/pulsar/data apachepulsar/pulsar:2.2.0 bin/pulsar standalone`

I got this error below:

```
08:20:45.537 [main] INFO  org.apache.bookkeeper.bookie.storage.ldb.DbLedgerStorage - Creating single directory db ledger storage on data/standalone/bookkeeper0/current
08:20:45.802 [main] INFO  org.apache.bookkeeper.proto.BookieNettyServer - Shutting down BookieNettyServer
Exception in thread "main" java.io.IOException: Error open RocksDB database
        at org.apache.bookkeeper.bookie.storage.ldb.KeyValueStorageRocksDB.<init>(KeyValueStorageRocksDB.java:175)
        at org.apache.bookkeeper.bookie.storage.ldb.KeyValueStorageRocksDB.<init>(KeyValueStorageRocksDB.java:79)
        at org.apache.bookkeeper.bookie.storage.ldb.KeyValueStorageRocksDB.lambda$static$0(KeyValueStorageRocksDB.java:54)
        at org.apache.bookkeeper.bookie.storage.ldb.LedgerMetadataIndex.<init>(LedgerMetadataIndex.java:70)
        at org.apache.bookkeeper.bookie.storage.ldb.SingleDirectoryDbLedgerStorage.<init>(SingleDirectoryDbLedgerStorage.java:170)
        at org.apache.bookkeeper.bookie.storage.ldb.DbLedgerStorage.newSingleDirectoryDbLedgerStorage(DbLedgerStorage.java:126)
        at org.apache.bookkeeper.bookie.storage.ldb.DbLedgerStorage.initialize(DbLedgerStorage.java:112)
        at org.apache.bookkeeper.bookie.Bookie.<init>(Bookie.java:734)
        at org.apache.bookkeeper.proto.BookieServer.newBookie(BookieServer.java:131)
        at org.apache.bookkeeper.proto.BookieServer.<init>(BookieServer.java:100)
        at org.apache.pulsar.zookeeper.LocalBookkeeperEnsemble.runBookies(LocalBookkeeperEnsemble.java:273)
        at org.apache.pulsar.zookeeper.LocalBookkeeperEnsemble.startStandalone(LocalBookkeeperEnsemble.java:385)
        at org.apache.pulsar.PulsarStandalone.start(PulsarStandalone.java:265)
        at org.apache.pulsar.PulsarStandaloneStarter.main(PulsarStandaloneStarter.java:102)
Caused by: org.rocksdb.RocksDBException: While fsync: a directory: Invalid argument
        at org.rocksdb.RocksDB.open(Native Method)
        at org.rocksdb.RocksDB.open(RocksDB.java:231)
        at org.apache.bookkeeper.bookie.storage.ldb.KeyValueStorageRocksDB.<init>(KeyValueStorageRocksDB.java:172)
        ... 13 more
08:20:51.838 [SessionTracker] INFO  org.apache.zookeeper.server.ZooKeeperServer - Expiring session 0x100011749cd0000, timeout of 6000ms exceeded
08:20:51.838 [ProcessThread(sid:0 cport:2181):] INFO  org.apache.zookeeper.server.PrepRequestProcessor - Processed session termination for sessionid: 0x100011749cd0000
```

**Comment by sijie** [comment]
Can you try to remove “ -v d:/temp/data:/pulsar/data ” to use non-mounted directory? I think the problem is how docker handle volumes on windows.

**Comment by kaonick** [comment]
I had tried it.
That is no problem.
Yes, I think is volumn handle error in windows.
So need for help to solve it ?

**Comment by sijie** [comment]
@kaonick great to hear that helps. I think the problem is a problem either at docker side or rocksdb side. I don't have any immediate solution to fix that. will have to search around to see if there is any solution to fix it.

**Comment by XuQianJin-Stars** [comment]
@kaonick I have submitted a PR, you can try again .
[https://github.com/apache/pulsar/pull/3198](https://github.com/apache/pulsar/pull/3198)

**Comment by sijie** [comment]
This is fixed by #3198
